shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Core.hs

```diff
@@ -14,46 +16,61 @@ module Cardano.Ledger.Core
   ( -- * Compactible
     Compactible (..),
     Compact (..),
-    Value,
+    Value (..),
+    VALUE,
   )
 where
 
 import Cardano.Binary (FromCBOR (..), ToCBOR (..))
+import Cardano.Ledger.Compactible
+import Cardano.Ledger.Val (Val)
+import Control.DeepSeq (NFData)
+import Data.Group (Abelian, Group)
 import Data.Kind (Type)
+import Data.PartialOrd (PartialOrd (..))
 import Data.Typeable (Typeable)
+import NoThunks.Class (NoThunks)
 
 -- | A value is something which quantifies a transaction output.
-type family Value era :: Type
+type family VALUE era :: Type
 
---------------------------------------------------------------------------------
+-- Wrap the type family as a newtype because :
+-- the genericShrink has something that
+-- detects that the immediate subterms of a type are the same as the parent type
+-- when there is a type family in that position, the instance resolution fails
+newtype Value era = Value {unVl :: VALUE era}
```

Comment:
I think it is because you have put `ShelleyEra c` where it was just `era` before, together with the `ShelleyBased era` constraint. `ShelleyEra` includes `Core.Value~Coin`, so `genericShrink` was not confused any more

Reply:
maybe rewrite the shrinker to a manual shrinker? to avoid this

Reply:
leaving shrinker as is for now